File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = 0
    
    ## 최대공약수
    def gcd(a, b):
        while b:
            a, b = b, a%b
        return a
    
    num1 = arr[0]
    for i in range(1, len(arr)):
        num2 = arr[i]
        
        num1 = num1*num2 // gcd(num1, num2)
        
    answer = num1
    
    return answer

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
        
        for(int n: arr){
            answer = (answer*n) / gcd(answer, n);
        }
        
        return answer;
    }
    
    public int gcd(int a, int b){
        while(b != 0){
            int tmp = a;
            a = b;
            b = tmp%b;
        }
        return a;
    }
}